### PR TITLE
Implement responsive actions bar

### LINKED ISSUE #1064: Custom rules for making actions bar buttons switch to 100% blocks at small widths needs implementing
**Describe the bug**
There are scratch rules applied to CXM and CMS which make actions buttons switch to 100% wide block layout when using mobile responsive width.

This behaviour needs bringing into Pulsar, or re-implementing with some flex magic given the upcoming change in browser support.

**Scratch rule**
```
.actionsbar {
    .float--left {
        float: none;

        @include respond-min($screen-smaller) {
            float: left;
        }
    }

    .float--right {
        float: none;

        @include respond-min($screen-smaller) {
            float: right;
        }
    }
}

.actionsbar > .float--left,
.actionsbar > .float--right {
    .dropdown {
        margin-bottom: 12px;

        @include respond-min($screen-smaller) {
            margin-bottom: 0;
        }

        > .btn {
            margin-bottom: 0;
        }
    }

    .btn__group {
        width: 100%;

        @include respond-min($screen-smaller) {
            width: auto;
        }

        .dropdown__menu {
            margin-bottom: 0;
            text-align: center;
            width: 100%;

            @include respond-min($screen-smaller) {
                text-align: left;
                width: inherit;
            }
        }

        > .btn__group:first-child {
            border-right: 0;
            border-radius: 4px;

            @include respond-min($screen-smaller) {
                border-bottom-right-radius: 0;
                border-top-right-radius: 0;
                border-right: 1px solid #ccc //use var
            }

            &:not(:last-child) .btn.dropdown__toggle {
                border-radius: 4px;

                @include respond-min($screen-smaller) {
                    border-bottom-right-radius: 0;
                    border-top-right-radius: 0;
                }
            }
        }

        > .btn__group:last-child {
            border-left: 0;
            border-radius: 4px;

            @include respond-min($screen-smaller) {
                border-bottom-left-radius: 0;
                border-top-left-radius: 0;
                border-left: 1px solid #ccc //use var
            }

            &:not(:first-child) .btn.dropdown__toggle {
                border-radius: 4px;

                @include respond-min($screen-smaller) {
                    border-bottom-left-radius: 0;
                    border-top-left-radius: 0;
                }
            }
        }

        .btn--inverse {
            border-left: 1px solid #333333;
            border-right: 1px solid #333333;

            &:first-child:not(:last-child):not(.dropdown__toggle) {
                @include respond-min($screen-smaller) {
                    border-right: 1px solid #333333;
                }
            }

            &:last-child:not(:first-child) {
                @include respond-min($screen-smaller) {
                    border-left: 1px solid #333333;
                }
            }

            .caret {
                border-top: 4px solid #fff;
            }
        }

        > a.btn:first-child {
            border-radius: 4px;

            @include respond-min($screen-smaller) {
                border-bottom-right-radius: 0;
                border-top-right-radius: 0;
            }
        }

        > a.btn:last-child {
            border-radius: 4px;

            @include respond-min($screen-smaller) {
                border-bottom-left-radius: 0;
                border-top-left-radius: 0;
            }
        }
    }

    .btn {
        display: block;
        margin-bottom: 12px;
        width: 100%;

        @include respond-min($screen-smaller) {
            display: inline-block;
            margin-bottom: 0;
            width: auto;
        }
    }
}
```

**Screenshot**
Screenshot 2019-11-18 at 14 11 37
Screenshot 2019-11-18 at 14 11 47

### PR DESCRIPTION
**Pull request by Stanton**
This change implements the current responsive actions bar styles which originated in CXM and is currently scratched in multiple products.

Closes #1064 

**Before**
Screenshot 2019-11-27 at 12 07 59

**After**
Screenshot 2019-11-27 at 12 07 24

### CODE REVIEW

**Comment by Stanton** [comment]
> Found an issue whilst testing where button group border radius isn't correct on mobile:
> 
> Screenshot 2019-11-28 at 10 57 03

Fixed:

Screenshot 2020-01-02 at 14 29 51

**Comment by jackbentley** [comment]
Looks like the middle button has a slight border on the left and right - and right has a slight border on the left?

**Comment by Stanton** [comment]
> Looks like the middle button has a slight border on the left and right - and right has a slight border on the left?

Good catch, thank you. Thee was also a -1px shift I needed to remove.

Screenshot 2020-01-03 at 08 15 43

File: stylesheets/_component.actionsbar.scss [hunk 1]
```diff
@@ -1,4 +1,156 @@
...
+            border-radius: 4px;
+
+            @include respond-min($screen-smaller) {
+                border-bottom-right-radius: 0;
+                border-top-right-radius: 0;
+                border-right: 1px solid color(base, dark);
+            }
+
+            &:not(:last-child) .btn.dropdown__toggle {
+                border-radius: 4px;
```
Comment (jamesjacobs) [inline stylesheets/_component.actionsbar.scss]: `$border-radius` var

File: stylesheets/_component.actionsbar.scss [hunk 2]
```diff
@@ -1,4 +1,156 @@
...
+                @include respond-min($screen-smaller) {
+                    border-bottom-right-radius: 0;
+                    border-top-right-radius: 0;
+                }
+            }
+        }
+
+        > .btn__group:last-child {
+            border-left: 0;
+            border-radius: 4px;
```
Comment (jamesjacobs) [inline stylesheets/_component.actionsbar.scss]: `$border-radius` var

File: stylesheets/_component.actionsbar.scss [hunk 3]
```diff
@@ -1,4 +1,156 @@
...
+
+            @include respond-min($screen-smaller) {
+                border-bottom-left-radius: 0;
+                border-top-left-radius: 0;
+                border-left: 1px solid color(base, dark);
+            }
+
+            &:not(:first-child) .btn.dropdown__toggle {
+                border-radius: 4px;
```
Comment (jamesjacobs) [inline stylesheets/_component.actionsbar.scss]: `$border-radius` var

File: stylesheets/_component.actionsbar.scss [hunk 4]
```diff
@@ -1,4 +1,156 @@
...
+                }
+            }
+
+            .caret {
+                border-top: 4px solid color(white);
+            }
+        }
+
+        > a.btn:first-child {
+            border-radius: 4px;
```
Comment (jamesjacobs) [inline stylesheets/_component.actionsbar.scss]: `$border-radius` var

File: stylesheets/_component.actionsbar.scss [hunk 5]
```diff
@@ -1,4 +1,156 @@
...
+
+            @include respond-min($screen-smaller) {
+                border-bottom-right-radius: 0;
+                border-top-right-radius: 0;
+            }
+        }
+
+        > a.btn:last-child {
+            border-radius: 4px;
```
Comment (jamesjacobs) [inline stylesheets/_component.actionsbar.scss]: `$border-radius` var

**Pull request merged**